plenum/test/req_drop/test_req_drop_on_propagate_phase.py

```diff
@@ -0,0 +1,98 @@
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = 3600
+    yield tconf
+
+    tconf.OUTDATED_REQS_CHECK_ENABLED = OUTDATED_REQS_CHECK_ENABLED_OLD
+    tconf.OUTDATED_REQS_CHECK_INTERVAL = OUTDATED_REQS_CHECK_INTERVAL_OLD
+    tconf.PROPAGATES_PHASE_REQ_TIMEOUT = PROPAGATES_PHASE_REQ_TIMEOUT_OLD
+    tconf.ORDERING_PHASE_REQ_TIMEOUT = ORDERING_PHASE_REQ_TIMEOUT_OLD
+
+
+@pytest.fixture()
+def setup(txnPoolNodeSet, looper, sdk_pool_handle, sdk_wallet_client):
+    global initial_ledger_size
+    A, B, C, D = txnPoolNodeSet  # type: TestNode
+    delay(Propagate, frm=[B, C, D], to=A, howlong=howlong)
+    # Delay MessageRep by long simulating loss as if Propagate is missing, it
+    # is requested
+    A.nodeIbStasher.delay(msg_rep_delay(10 * howlong, [PROPAGATE, ]))
+    initial_ledger_size = txnPoolNodeSet[0].domainLedger.size
+    request_couple_json = sdk_send_random_requests(
+        looper, sdk_pool_handle, sdk_wallet_client, 1)
+    return request_couple_json
+
+
+def testReqDropOnPropagatePhaseOnMasterPrimaryAndThenOrdered(tconf, setup, looper, txnPoolNodeSet,
+                                                             sdk_wallet_client, sdk_pool_handle):
+    A, B, C, D = txnPoolNodeSet  # type: TestNode
+    sent1 = sdk_json_to_request_object(setup[0][0])
+
+    def check_propagates_delayed():
+        # A should have received a request from the client
+        assert len(recvdRequest(A)) == 1
+        # A should not have received a PROPAGATE
+        assert len(recvdPropagate(A)) == 0
+        # A should have sent a PROPAGATE
+        assert len(sentPropagate(A)) == 1
+        assert len(A.requests) == 1
+
+    timeout = howlong - 2
+    looper.run(eventually(check_propagates_delayed, retryWait=.5, timeout=timeout))
+
+    def check_drop():
+        assert len(A.requests) == 0
+
+    timeout = tconf.PROPAGATES_PHASE_REQ_TIMEOUT + tconf.OUTDATED_REQS_CHECK_INTERVAL + 1
+    looper.run(eventually(check_drop, retryWait=.5, timeout=timeout))
+
+    for n in txnPoolNodeSet:
+        n.nodeIbStasher.resetDelays()
+
+    def check_propagates_received():
+        # A should have received 3 PROPAGATEs
+        assert len(recvdPropagate(A)) == 3
+        # A should have total of 4 PROPAGATEs (3 from other nodes and 1 from
+        # itself)
+        key = sent1.digest
+        assert key in A.requests
+        assert len(A.requests[key].propagates) == 4
+        # A should still have sent two PROPAGATEs since request
+        # was dropped and re-received over propagate
+        assert len(sentPropagate(A)) == 2
+
+    timeout = howlong + 2
+    looper.run(eventually(check_propagates_received, retryWait=.5, timeout=timeout))
+
+    def check_ledger_size():
+        for node in txnPoolNodeSet:
+            assert node.domainLedger.size - initial_ledger_size == 1
+
+    looper.run(eventually(check_propagates_received, retryWait=.5, timeout=timeout))
+
+    sdk_ensure_pool_functional(looper, txnPoolNodeSet, sdk_wallet_client, sdk_pool_handle)
```

Comment:
It looks like we still don't check that the request is ordered eventually.
`sdk_ensure_pool_functional` sends a new request, and this is good to check this, we we need to make sure that the dropped request has been eventually ordered on all nodes.

Reply:
It's a mistake, there should be _check_ledger_size_ under _eventually()_.